main/src/cgeo/geocaching/connector/al/ALApi.java

```diff
@@ -383,8 +383,15 @@ private static Geocache parseCacheDetail(final JsonNode response) {
             cache.setHidden(parseDate(response.get("PublishedUtc").asText()));
             cache.setOwnerDisplayName(response.get("OwnerUsername").asText());
             cache.setWaypoints(parseWaypoints((ArrayNode) response.path("GeocacheSummaries")), true);
-            cache.setDetailedUpdatedNow();
+            final boolean isLinear = response.get("IsLinear").asBoolean();
+            if (isLinear) {
+                cache.setAlcMode(1);
+            } else {
+                cache.setAlcMode(0);
+            }
+            Log.e("_AL mode from JSON: IsLinear: " + cache.isLinearAlc());
             DataStore.saveCache(cache, EnumSet.of(SaveFlag.DB));
+            cache.setDetailedUpdatedNow();
```

Comment:
Not sure what impact this has, but why did you move this line _after_ `DataStore.saveCache(...)`?

Reply:
I felt if these lines are strictly executed synchronously that the term "...UpdatedNow" (past tense) is only really true **after** the `saveCache()` returns.

Reply:
Haven't checked, but doesn't `cache.setDetailedUpdateNow()` update some data fields which need to be persisted as well? Or are they in-memory fields only?